src/OrchardCore/OrchardCore.Data/Migration/DataMigrationManager.cs

```diff
@@ -166,7 +166,7 @@ public async Task UpdateAsync(string featureId)
                 var current = 0;
                 if (dataMigrationRecord != null)
                 {
-                    current = dataMigrationRecord.Version.Value;
+                    current = dataMigrationRecord.Version.HasValue ? dataMigrationRecord.Version.Value : current;
                 }
```

Comment:
This gets saved with a null value if the migration has an exception and it's the `Create()` migration.

So adding this doesn't exactly fix the problem, but it allows the migration to be fixed, and then run again